### DISCUSSION TITLE
Our android port (Sherpa) is on Google play 👍

### DISCUSSION THREAD

**Discussion by ThibautLEAUX** [ggml-org/llama.cpp#750] [General]
Hello everyone !

Our android port of llama.cpp is on [Google Play](https://play.google.com/store/apps/details?id=com.biprep.sherpa) !

The repo is here : [https://github.com/Bip-Rep/sherpa](https://github.com/Bip-Rep/sherpa), we hope that more people can work on it because we are really amazed that it can run on a mobile (with at least 8GB RAM). 
Flutter devs, feel free to make PRs.

## Working demo
[![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/jdw7oABjTeQ/0.jpg)](https://www.youtube.com/watch?v=jdw7oABjTeQ)

Click on the image to view the video on YouTube.
It shows a OnePlus 7 with 8Gb running Sherpa without speed up.

**Comment by MillionthOdin16** [comment]
Awesome! I've been waiting :)

**Reply by MillionthOdin16** [reply]
None of my models work :( does any 7B model that works in llama.cpp work?

**Reply by ThibautLEAUX** [reply]
You need to convert the models with the scripts available in llama.cpp repo.
You must use [convert-unversioned-ggml-to-ggml.py](https://github.com/ggerganov/llama.cpp/blob/master/convert-unversioned-ggml-to-ggml.py) and it should be working with the app !

**Reply by MillionthOdin16** [reply]
Hmmm, I'm getting magic number errors. I've tried with multiple versions of the model (like alpaca, gpt4all) and with multiple versions of the ggml implementation. Before and after the nmap integration as well.

**Reply by tibzejoker** [reply]
@MillionthOdin16 could you try again ? it should work super fine with vicuna now :)

**Reply by MillionthOdin16** [reply]
I still get force closes :/

**Reply by ThibautLEAUX** [reply]
Could you tell me how much ram you have on your device please?

**Reply by MillionthOdin16** [reply]
12gb. Pixel 6 pro

**Reply by krecicki** [reply]
this is the same thing for me every time

**Comment by ivanstepanovftw** [comment]
Same for me, getting bad magic

**Reply by tibzejoker** [reply]
@ivanstepanovftw we updated it if you want to try

**Comment by Cryptoris-Tech** [comment]
Can I build llama.cpp binary for 32 bit? 
Or is there any precompiled binary for 32 bit?
